backend_jobs/recognition_pipeline/pipeline_lib/redefine_labels.py

```diff
@@ -0,0 +1,50 @@
+"""
+  Copyright 2020 Google LLC
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+"""
+
+import apache_beam as beam
+from pipeline_lib.firestore_database import initialize_db
+
+# pylint: disable=missing-function-docstring
+def get_redefine_map(recognition_provider_id):
+    db = initialize_db()
+    doc_dict = db.collection(u'RedefineMaps').document(recognition_provider_id).get().to_dict()
+    return doc_dict['redefineMap']
+
+# pylint: disable=abstract-method
+class RedefineLabels(beam.DoFn):
+    """Converts parallelly the labels list returned from
+    the provider to the corresponding label Id's.
+
+    """
+    # pylint: disable=arguments-differ
+    def process(self, element, provider_id):
+        """Uses the global redefine map to change the different labels to the project's label Ids.
+
+        Args:
+            element: tuple of dictionary of image properties and list of labels.
+            provider_id: image recognition provider for the redefine map.
+
+        Returns:
+            [(dictionary of image properties, label ids list)]
+        """
+        all_labels_and_ids = []
+        for label in element[1]:
+            redefine_map = get_redefine_map(provider_id)
```

Comment:
this approach is not efficient. for each label   it will do a DB lookup for the map.... though this config will never change for a given pipeline